packages/world/src/utils/callChip.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import { ResourceId } from "@latticexyz/store/src/ResourceId.sol";
+import { WorldContextConsumerLib, WorldContextProviderLib } from "@latticexyz/world/src/WorldContext.sol";
+import { revertWithBytes } from "@latticexyz/world/src/revertWithBytes.sol";
+import { Systems } from "@latticexyz/world/src/codegen/tables/Systems.sol";
+
+import { Chip } from "../codegen/tables/Chip.sol";
+import { EntityId } from "../EntityId.sol";
+import { SAFE_CHIP_GAS } from "../Constants.sol";
+
+function callChip(address chipAddress, bytes memory callData) returns (bool, bytes memory) {
+  address msgSender = WorldContextConsumerLib._msgSender();
+  uint256 msgValue = WorldContextConsumerLib._msgValue();
+
+  return
+    chipAddress.call{ value: 0 }(
```

Comment:
With chips being systems, I thought we'd call them through the world vs the system address directly?

Reply:
We are calling it through the world (as we only should use these calls from root systems) but appending the world context manually